### PR TITLE
[CFE-477]: Feature/cfe 477 txs by addr lcd

### PR DESCRIPTION
**Pull request by Poafs1**

### CODE REVIEW

File: src/lib/services/tx/index.ts [hunk 1]
```diff
@@ -53,15 +53,17 @@ export const useTxData = (
...
-        ...txResponse,
+        ...payload,
         timestamp: parseDateOpt(txResponse.timestamp),
         logs,
         chainId: currentChainId,
         isTxFailed: Boolean(txResponse.code),
       };
     },
-    [endpoint, currentChainId, isFullTier]
+    [currentChainId, endpoint, isFullTier]
```
Comment (evilpeach) [inline src/lib/services/tx/index.ts:66]: Only endpoint should be enough.

File: src/lib/services/tx/lcd.ts [hunk 2]
```diff
@@ -1,6 +1,30 @@
...
+    )
+    .then(({ data }) => parseWithError(zTxsByAddressResponseLcd, data));
+
+export const getTxsByAccountAddressLcd = async (
+  endpoint: string,
+  address: BechAddr20
+) =>
+  axios
+    .get(
+      `${endpoint}/cosmos/tx/v1beta1/txs?events=message.sender=%27${encodeURI(address)}%27&order_by=2`
```
Comment (evilpeach) [inline src/lib/services/tx/lcd.ts:28]: apply here as well

File: CHANGELOG.md [hunk 3]
```diff
@@ -39,6 +39,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Features
 
+- [#923](https://github.com/alleslabs/celatone-frontend/pull/923) Add get txs by contact and account addr lcd
+- [#920](https://github.com/alleslabs/celatone-frontend/pull/920) Support lite version contract with new folder structure
```
Comment (evilpeach) [inline CHANGELOG.md:43]: I think 920 shouldn't be included here.

File: src/lib/services/tx/index.ts (hunk 1 above)
Comment (Poafs1) [reply src/lib/services/tx/index.ts:66]: If we use only `endpoint` parameter, the Eslint will report the error.

File: src/lib/services/types/tx.ts [hunk 4]
```diff
@@ -17,72 +19,110 @@ import {
...
-  "@type": TypeUrl;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  [key: string]: any;
-}
+const zTxBody = z
+  .object({
+    messages: z.array(zMessageResponse).nullable(),
+    memo: z.string(),
+    timeout_height: z.string(),
+    extension_options: z.any().array(),
```
Comment (evilpeach) [inline src/lib/services/types/tx.ts:71]: Create zAny for this that convert to `import type { Any } from "cosmjs-types/google/protobuf/any";`

File: src/lib/services/types/tx.ts [hunk 5]
```diff
@@ -17,72 +19,110 @@ import {
...
+});
+export type Event = z.infer<typeof zEvent>;
+
+const zLog = z
+  .object({
+    msg_index: z.number(),
+    log: z.string(),
+    events: z.array(zEvent),
+  })
+  .transform(snakeToCamel);
```
Comment (evilpeach) [inline src/lib/services/types/tx.ts:105]: I think it's no need to apply `snakeToCamel` here, @songwongtp What do you think?

File: src/lib/services/types/proposal.ts [hunk 6]
```diff
@@ -134,13 +135,7 @@ export const zProposalDataResponse = z.object({
...
       created_tx_hash: z.string().nullable(),
       description: z.string(),
-      messages: z
-        .object({
-          "@type": z.string(),
-        })
-        .passthrough()
-        .array()
-        .nullable(),
+      messages: z.array(zMessageResponse).nullable(),
```
Comment (songwongtp) [inline src/lib/services/types/proposal.ts:138]: Remove nullable

File: src/lib/utils/tx/extractTxLogs.ts [hunk 7]
```diff
@@ -1,25 +1,29 @@
...
   if (txData.logs.length > 0) return txData.logs;
 
   // post Cosmos SDK 0.50
-  const msgLogs = txData.tx.body.messages.map((_, index) => ({
-    msg_index: index,
-    log: "",
-    events: [] as Event[],
-  }));
+  const msgLogs =
+    txData.tx.body.messages?.map((_, index) => ({
```
Comment (songwongtp) [inline src/lib/utils/tx/extractTxLogs.ts:14]: Remove ?

**Review by songwongtp — CHANGES_REQUESTED** [review]

**Review by songwongtp — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**